app/models/teacher.rb

```diff
@@ -0,0 +1,35 @@
+class Teacher < ActiveRecord::Base
+
```

Comment:
I like your initiative to add validations. Would it be difficult to remove the password and session related changes to another PR?